instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/test/groovy/WithCurrentSpanInstrumentationTest.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import io.opentelemetry.test.annotation.SpanAttributesWithCurrentSpan
+
+import static io.opentelemetry.api.trace.SpanKind.INTERNAL
+
+/**
+ * This test verifies that auto instrumentation supports the
+ * {@link io.opentelemetry.instrumentation.annotations.WithCurrentSpan}
+ * and
+ * {@link io.opentelemetry.instrumentation.annotations.SpanAttribute}
+ * annotations.
+ */
+class WithCurrentSpanInstrumentationTest extends AgentInstrumentationSpecification {
```

Comment:
Can you refactor this into a Java test class? We're slowly trying to convert all the Groovy tests into Java (#7195) and we're trying not to introduce too much groovy code (unless it's straight copy-paste, e.g. when instrumenting a new version of an already instrumented library)